x-pack/plugins/security_solution/scripts/run_cypress/parallel_serverless.ts

```diff
@@ -0,0 +1,508 @@
+): Promise<Credentials> {
+  console.log(`${runnerId} : Reseting credentials`);
+  let credentials = {} as Credentials;
+
+  await poll(async () => {
+    return await axios
+      .post(
+        `${baseUrl}/api/v1/serverless/projects/security/${environmentId}/_reset-credentials`,
+        {},
+        {
+          headers: {
+            Authorization: `ApiKey ${apiKey}`,
+          },
+        }
+      )
+      .then((response) => {
+        credentials.password = response.data.password;
+        credentials.username = response.data.username;
+      })
+      .catch((error) => {
+        throw error;
+      });
+  });
+  return credentials;
+}
+
+// Wait until elasticsearch status goes green
+async function waitForEsStatusGreen(esUrl: string, auth: string, runnerId: string) {
+  await poll(async () => {
+    await axios
+      .get(`${esUrl}/_cluster/health?wait_for_status=green&timeout=50s`, {
+        headers: {
+          Authorization: `Basic ${auth}`,
+        },
+      })
+      .then((response) => {
+        console.log(`${runnerId}: Elasticsearch is ready with status ${response.data.status}.`);
+      })
+      .catch((error) => {
+        if (error.code == 'ENOTFOUND') {
+          console.log(
+            `${runnerId}: The elasticsearch url is not yet reachable. Retrying in 20s...`
+          );
+        }
+        throw error;
+      });
+  });
+}
+
+// Wait until Kibana is available
+async function waitForKibanaAvailable(kbUrl: string, auth: string, runnerId: string) {
+  await poll(async () => {
+    await axios
+      .get(`${kbUrl}/api/status`, {
+        headers: {
+          Authorization: `Basic ${auth}`,
+        },
+      })
+      .then((response) => {
+        if (response.data.status.overall.level != 'available') {
+          console.log(`${runnerId}: Kibana is not available. Retrying in 20s...`);
+          throw new Error(`${runnerId}: Kibana is not available. Retrying in 20s...`);
+        }
+      })
+      .catch((error) => {
+        if (error.code == 'ENOTFOUND') {
+          console.log(`${runnerId}: The kibana url is not yet reachable. Retrying in 20s...`);
+        }
+        throw error;
+      });
+  });
+}
+
+export const cli = () => {
+  run(
+    async () => {
+      const log = new ToolingLog({
+        level: 'info',
+        writeTo: process.stdout,
+      });
+
+      const PROJECT_NAME_PREFIX = 'kibana-cypress-security-solution-ephemeral';
+      const QA_BASE_ENV_URL = 'https://global.qa.cld.elstc.co';
+
+      // Checking if API key is either provided via env variable or in ~/.elastic.cloud.json
+      if (!process.env.API_KEY && !getApiKeyFromElasticCloudJsonFile()) {
+        log.error('The api key for the environment needs to be provided with the env var API_KEY.');
+        log.error(
+          'If running locally, ~/.elastic/cloud.json is attempted to be read which contains the api key.'
+        );
+        return process.exit(0);
+      }
+
+      const API_KEY = process.env.API_KEY
```

Comment:
I would target a more specific environment variable name to avoid name conflicts with other, unrelated API keys.

Reply:
Fixed by [65951be28d55b77bfa3721811841307d05f8ac7d](https://github.com/elastic/kibana/commit/65951be28d55b77bfa3721811841307d05f8ac7d)